fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/properties/PrimaryKeyProperty.java

```diff
@@ -312,6 +313,9 @@ public Optional<List<Value>> visitUnorderedUnionPlan(@Nonnull final RecordQueryU
         @Nonnull
         @Override
         public Optional<List<Value>> visitScanPlan(@Nonnull final RecordQueryScanPlan scanPlan) {
+            if (Type.anyRecord().equals(scanPlan.getResultType().getInnerType())) {
+                return Optional.empty();
+            }
```

Comment:
without this "if" statement, we could try to translate primary key expression on AnyRecord and fail.

Reply:
Could this cause the common primary key shared of two types to be incorrectly not returned?